pkg/cloud/aws/aws.go

```diff
@@ -136,38 +139,15 @@ func (a *awsProvider) PrepareSubmarinerClusterEnv() error {
 
 // CleanUpSubmarinerClusterEnv clean up submariner cluster environment on AWS after the SubmarinerConfig was deleted
 func (a *awsProvider) CleanUpSubmarinerClusterEnv() error {
-	if err := a.gatewayDeployer.Cleanup(a); err != nil {
+	if err := a.gatewayDeployer.Cleanup(a.reporter); err != nil {
 		return err
 	}
 
-	if err := a.cloudPrepare.CleanupAfterSubmariner(a); err != nil {
+	if err := a.cloudPrepare.CleanupAfterSubmariner(a.reporter); err != nil {
 		return err
 	}
 
 	a.eventRecorder.Eventf("SubmarinerClusterEnvCleanedUp", "the submariner cluster env is cleaned up on aws")
```

Comment:
We could use the `recorder` wrapper here and in `PrepareSubmarinerClusterEnv` for consistency, then we can remove the `eventRecorder` field. Same in the GCP code.